### PR TITLE
{App Service} Upgrade Microsoft.Web to v2023-01-01

### PR DESCRIPTION
**Pull request by kamperiadis**
**Related command**

**Description**

**Testing Guide**

**History Notes**

Upgrade Microsoft.Web to v2023-01-01

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
AppService

**Comment by yonzhan** [comment]
Please fix CI issues.

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 1]
```diff
@@ -3028,9 +3028,9 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
         if kv_subscription.lower() != subscription_id.lower():
             diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
                                                                subscription_id=kv_subscription)
-            ascs = diff_subscription_client.app_service_certificate_orders.list()
+            ascs = diff_subscription_client.app_service_certificate_orders.list(api_version='2022-09-01')
         else:
-            ascs = client.app_service_certificate_orders.list()
+            ascs = client.app_service_certificate_orders.list(api_version='2022-09-01')
```
Comment (yanzhudd) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3033]: It is not recommended to hard code the api_version in custom.py file. You can add ad-hoc api-version in https://github.com/Azure/azure-cli/blob/dev/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L432.

Hi @jiasli, do you have any suggestions for this?

Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Thanks for your comment @yanzhudd. I just addressed it. Can you please review the PR again? Thank you!

File: src/azure-cli-core/azure/cli/core/profiles/_shared.py [hunk 2]
```diff
@@ -430,6 +430,9 @@ def default_api_version(self):
 # We should avoid using ad hoc API versions,
 # use the version in a profile as much as possible.
 AD_HOC_API_VERSIONS = {
+    ResourceType.MGMT_APPSERVICE: {
+        'app_service_certificate_orders': '2022-09-01'
+    },
```
Comment (jiasli) [inline src/azure-cli-core/azure/cli/core/profiles/_shared.py:435]: Please add a comment showing where it is used, such as 

https://github.com/Azure/azure-cli/blob/08e55c47ff77e2aa8f87cfd18c5d3bcaa52d2c9d/src/azure-cli-core/azure/cli/core/profiles/_shared.py#L434

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py [hunk 3]
```diff
@@ -3023,14 +3023,17 @@ def import_ssl_cert(cmd, resource_group_name, name, key_vault, key_vault_certifi
...
     from azure.cli.core.commands.client_factory import get_subscription_id
+    from azure.cli.core.profiles import AD_HOC_API_VERSIONS
     subscription_id = get_subscription_id(cmd.cli_ctx)
     if cloud_type.lower() == PUBLIC_CLOUD.lower():
         if kv_subscription.lower() != subscription_id.lower():
             diff_subscription_client = get_mgmt_service_client(cmd.cli_ctx, ResourceType.MGMT_APPSERVICE,
                                                                subscription_id=kv_subscription)
-            ascs = diff_subscription_client.app_service_certificate_orders.list()
+            ascs = diff_subscription_client.app_service_certificate_orders.list(
+                api_version=AD_HOC_API_VERSIONS[ResourceType.MGMT_APPSERVICE]['app_service_certificate_orders'])
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/appservice/custom.py:3033]: `AD_HOC_API_VERSIONS` was introduced to preserve certain API versions while trimming mgmt SDKs (#17816, #23946).

As a side note, though your current code is better than hard-coding API version `2022-09-01` here, there is no hard requirement that you must use an ad-hoc API version as `AD_HOC_API_VERSIONS[ResourceType.MGMT_APPSERVICE]['app_service_certificate_orders']`.

For example, `acr` just hard-codes them:

https://github.com/Azure/azure-cli/blob/09a6fff1a34447dbf0987621ca89ab1c1911529d/src/azure-cli/azure/cli/command_modules/acr/_client_factory.py#L8-L13

#25418 also hard-codes `2022-01-31-preview`:

https://github.com/Azure/azure-cli/blob/cbb33a92a291e86e877ff199a738fe8ec17acda2/src/azure-cli/azure/cli/command_modules/identity/_client_factory.py#L18

It is acceptable to use `2022-09-01` here, as long as the hard-coded API version is the same as what is defined in `AD_HOC_API_VERSIONS`.

So, no action is needed! 🙂

**Comment by yanzhudd** [comment]
please fix the CI failed items.

**Review by yanzhudd — DISMISSED** [review]

File: src/azure-cli/azure/cli/command_modules/appservice/custom.py (hunk 1 above)
Comment (zhoxing-ms) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py:3033]: Suggest defining a reusable variable for `api_version='2022-09-01'`
```
 ResourceType.MGMT_APPSERVICE: {
        # src/azure-cli/azure/cli/command_modules/appservice/custom.py:3033
        'app_service_certificate_orders': '2022-09-01'
    },
```
`# src/azure-cli/azure/cli/command_modules/appservice/custom.py:3033` so the referenced code link can point to reusable variable

Comment (kamperiadis) [reply src/azure-cli/azure/cli/command_modules/appservice/custom.py]: Ok I have addressed your comment. Can you please review it again @zhoxing-ms and merge if it looks good? Thank you!

**Review by necusjz — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**